tools/cleanup-buildenv-build-dir.py

```diff
@@ -171,15 +202,49 @@ def removeFilesInDir(targetDir):
 				rm(absFn)
 
 def main():
-	if len(sys.argv) < 2:
-		print('Usage: python cleanup-buildenv-build-dir.py <build-dir>')
-		print('')
-		print(' For example:')
-		print('   python cleanup-buildenv-build-dir.py /MumbleBuild/centos-ermine-1.2.x-2014-06-01-cf59267.build')
-		sys.exit(1)
-
-	build_dir = sys.argv[1]
+	global verbose
+	global dryrun
+	global KEEP_EXT
+	
+	parser = argparse.ArgumentParser(description='Cleans up the <buildenv-name>.build directories that are used for storing source zips, tarballs and source trees when a build env is being built.')
+	parser.add_argument('builddir', help='<buildenv-name>.build directory (e.g. c:/MumbleBuild/centos-ermine-1.2.x-2014-06-01-cf59267.build)')
+	parser.add_argument('--dry-run', action='store_true')
+	parser.add_argument('-v', '--verbose', action='store_true')
+	parser.add_argument('--force' , action='store_true', help = 'Force run on directories not ending in .build')
+	args = parser.parse_args();
+	verbose = args.verbose
+	dryrun = args.dry_run
+	build_dir = args.builddir
+	
+	if verbose: print("Performing cleanup on " + build_dir)
+	if dryrun: print("Performing dry run. No files will actually be deleted.")
+	
+	if not args.force:
+		build_dirname =  os.path.basename(os.path.normpath(build_dir))
+		if not build_dirname.endswith(".build"):
+			print("Build directory name '" + build_dirname + "' does not end in .build . If you are sure you want to clean this path re-execute with --force")
+			sys.exit(1)
+	
+	# Perform initial cleanup
+	if verbose: print("Performing initial cleanup")
 	removeFilesInDir(build_dir)
-
+	if verbose: print("Done with initial cleanup")
+	
+	# After that perform additional cleanup
+	if verbose: print("Performing additional cleanup")
+	# !!!!!
+	KEEP_EXT = [] # !!! We do not obey preversed extensions after this point !!!
```

Comment:
How about we add a keep_ext parameter to removeFilesInDir to avoid the global, and this scary block?